### PR TITLE
chore: update Prettier

### PR DESCRIPTION
**Pull request by P0lip**
Related to https://github.com/stoplightio/spectral/issues/1152

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Additional context**

The `.prettierrc` config is taken from https://github.com/stoplightio/eslint-config/blob/master/prettier.config.js

Note, the resolution won't be needed as soon as we drop tslint-config-stoplight that depends on the older version.

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -2,5 +2,8 @@
   "extends": ["tslint-config-stoplight"],
   "linterOptions": {
     "exclude": ["**/*.json"]
+  },
+  "rules": {
+    "prettier": "error"
```
Comment (P0lip) [inline tslint.json:7]: This makes sure we don't use the config enforced by tslint-config-stoplight, but choose the one we have in prettierc instead.

File: package.json [hunk 2]
```diff
@@ -41,8 +41,9 @@
     "cli:debug": "node -r ts-node/register -r tsconfig-paths/register --inspect-brk src/cli/index.ts",
     "generate-assets": "node -r ts-node/register ./scripts/generate-assets.ts",
     "inline-version": "./scripts/inline-version.js",
+    "prelint": "tsc --noEmit",
     "lint.fix": "yarn lint --fix",
-    "lint": "tsc --noEmit && tslint 'src/**/*.ts'",
```
Comment (nulltoken) [inline package.json:45]: @P0lip Do we really need to drop that typechecking step?

Comment (P0lip) [reply package.json:45]: It's moved to prelint. Prelint is always executed before lint, so we don't lose anything.
Type checking is indeed necessary.

Comment (nulltoken) [reply package.json:45]: Perfect! I overlooked that. Thanks!

**Pull request merged**